### ISSUE TITLE
Speed up with larger libraries?

### ISSUE THREAD

**Issue by Mike120** [boxeehacks/boxeehack#190]
My media library has 200 shows with 6000 episodes and 2200 movies. Browsing through the movies & shows sections is extremely slow. I'll press the button and 1 or 2 seconds later it will move along.

Also selecting an episode takes a while and after clicking on it to play it takes between 10 & 20 seconds to actually load the video. Exiting the video and trying to go back proves impossible most of the time (only thing that works is pulling the plug out & rebooting).

Can better support be added for larger libraries?

**Comment by war59312** [comment]
I'm sure they will try, but hardware is very much a limitation with Boxee Box.

The CPU and RAM Boxee Box uses is very week compared to today's similar hardware.

I highly doubt much can be done about it. Specially since there is no real "cache" to speak of. Not enough RAM and currently NOT possible to store to "offsite" "RAM", think a USB thumb drive.

That being said, something is very much wrong. Since on my BB I now have more TV Shows and Movies than you and whiles its a bit slow its not that slow. More like 2 seconds to load a video and scrolling is not that bad.

All connections between devices are at GB speed. And I am using NFS and running on old 1.2 firmware. My BB hardware revision is the original, I think it's A1 (not at home to verify, and I know I stated correctly in the BB forums).

**Comment by skipperbent** [comment]
war59312 - could it be possible to make an NFS share into some kind of "virtual ram" like Windows does with it's page-file?

Because then you would have almost endless storage depending on your lan-setup.

**Comment by Mike120** [comment]
Yeah obviously the hardware is pretty outdated now but on regular boxee it's pretty 'fast' to do most things for me.

Hopefully they'll come up with something

**Comment by jairofsouza** [comment]
I have problems with my music library (1000 artists). Its very slow...

**Comment by war59312** [comment]
Skipper, not sure. I'm sure its being investigated. One of the top complaints about BB from day 1.

See: http://jira.boxee.tv/browse/BOXEE-7446 (Note how the bug report is now 2+ years old).

Jairofsouza, sorry don't use BB for music. So no experience there.

Mike, I'm afraid I would not hold me breath for now anyways. If compiling becomes a reality then I could see this fixed once and for all but until then. I doubt it.

**Comment by jesseblue** [comment]
i think the bbox is resizing the pics that you might be using for the poster...what dimensions do you have for the folder.jpg? only true if you have individual folder.jpgs

**Comment by Mike120** [comment]
@jesseblue I don't have folder.jpgs only .tbns...

**Comment by jesseblue** [comment]
how big is your tbn file?

**Comment by Mike120** [comment]
Average size is between 25-60kb

**Comment by FabianMeul** [comment]
What about using the SD-card slot as Caching location?

**Comment by jesseblue** [comment]
weird...i'm not sure whatcould be causing this....i don't use tbn files so i can't comment too much on it...i hope that other people who use this type of file can chime in...

**Comment by F33db3ck** [comment]
I think it's a good idea, i agree with FabianMeul. using the SD card slot as a caching location.

Everytime i open my movies, music or anything else, the thumbnails are loading again, it's a real bummer.

**Comment by war59312** [comment]
Mike120, you have limited the dimensions of the thumbnails right?

For movies it's 299 (width) by 434 (height).

For TV Show Posters it's also 299 (width) by 434 (height).

But for TV Episodes it's 299 (width) by 160 (height).

**Comment by Mike120** [comment]
@war59312 Not all of them are. I'd say 90% of the TV shows are limited and maybe 50-60% of Movies.

It has never been slow before boxee+ though, and it's not just when I go to browse that It's slow. Exiting a movie or show lags for minutes before It will even go back to the browser.

I've uninstalled it now as regular boxee is suprisngly faster and problem free (mostly) and will wait for further updates

**Comment by war59312** [comment]
OK got you, so an issue with the modifications. Well I suppose you can try a different older version of the hack and try and see when the problem first starts or if it starts from the very first build.
